nf_core/components/install.py

```diff
@@ -38,15 +41,33 @@ def __init__(
         installed_by: Optional[List[str]] = None,
     ):
         super().__init__(component_type, pipeline_dir, remote_url, branch, no_pull)
+        self.current_remote = ModulesRepo(remote_url, branch)
+        self.branch = branch
         self.force = force
         self.prompt = prompt
         self.sha = sha
+        self.current_sha = sha
```

Comment:
what is the difference between `self.sha` and `self.current_sha`?

Reply:
if I understand correctly, `current_sha` keeps track of the sha of all included components, while `sha` is always the one from the installed subworkflow. 
Since included componenets can be from different repos, the shas might differ form the original installed subworkflow one.

Reply:
That's correct!